Rename crate to parity-snappy

andresilva opened the pull request:
This is necessary for publishing to crates.io (a snappy crate already exists there).

andresilva committed:
rename crate to parity-snappy

andresilva committed:
run rustfmt

andresilva commented:
I also ran rustfmt on this and added a config file to use tabs instead of spaces.

dvdplm commented on snappy-sys/Cargo.toml:
```diff
@@ -1,7 +1,7 @@
+authors = ["Parity Technologies <admin@parity.io>"]
 license = "MIT/Apache-2.0/BSD"
 description = "Native bindings to libsnappy"
 readme = "README.md"
```
Adding a `repository` key is helpful.

dvdplm reviewed (approved):
lgtm modulo repository metadata key

andresilva committed:
add more metadata to Cargo.toml

andresilva committed:
publish pre-release version

andresilva committed:
license under GPL3

All of the source code already had headers licensing under GPLv3, but the repo
was missing the LICENSE file and the license was wrongly specified in Cargo.toml

andresilva committed:
update readme

andresilva committed:
fix typo in readme

andresilva committed:
release version v0.1.0
